src/Loading/sceneLoader.ts

```diff
@@ -479,7 +479,7 @@ export class SceneLoader {
 
         SceneLoader.OnPluginActivatedObservable.notifyObservers(plugin);
 
-        if (directLoad) {
+        if (directLoad && plugin.canDirectLoad && plugin.canDirectLoad(fileInfo.url)) {

```

Comment:
This change is currently breaking the scenario (and unit test) for OBJ files being passed in as raw text by specifying the data as "data:<obj_file_content>"

I didn't realize that was a supported scenario because the OBJ plugin does not declare the `canDirectLoad` or `directLoad` functions. In the previous implementation if the data was a string that started with "data:" it would fallback stripping "data:" from the start of the string and passing the rest of the value to the plugins default loader method.

Reply:
I've resolved the regression by updating the conditional to still use the directLoad code path if the given data URI is not a valid base64 data uri. That retains the pre-existing support for non standard "data:" urls while still making it so loaders can remain oblivious to base64 data urls.

The one gotcha I found was that we still need to use the glTF directLoad function when it registers as a handler for base64 URI's. That's because it's only in it's directLoad function that it detects if it's a gltf or glb file.